tokendito/okta_helpers.py

```diff
@@ -20,30 +20,41 @@
 
 from future import standard_library
 import requests
+from tokendito import duo_helpers
 from tokendito import helpers
 from tokendito import settings
+
 standard_library.install_aliases()
 
 
-def okta_verify_api_method(mfa_challenge_url, payload, headers):
+def okta_verify_api_method(mfa_challenge_url, payload, headers=None):
     """Okta MFA authentication.
 
     :param mfa_challenge_url: MFA challenge url
     :param payload: JSON Payload
     :param headers: Headers of the request
     :return: Okta authentication response
     """
-    logging.debug("Okta MFA authentication URL [{}] headers [{}]".format(
-        mfa_challenge_url, headers))
-
     try:
-        response = json.loads(requests.request('POST', mfa_challenge_url,
-                                               data=json.dumps(payload), headers=headers).text)
+        if headers:
+            response = requests.request('POST', mfa_challenge_url,
+                                        data=json.dumps(payload), headers=headers)
+        else:
+            response = requests.request(
+                'POST', mfa_challenge_url, data=payload)
     except Exception as request_error:
         logging.error(
             "There was an error connecting to Okta: \n{}".format(request_error))
         sys.exit(1)
 
+    logging.debug("Okta authentication response: \n{}".format(response))
+
+    try:
+        response = response.json()
+    except ValueError:
+        logging.debug("Received type of response: {}".format(type(response.text)))
+        response = response.text
+
     if 'errorCode' in response:
         error_string = "Exiting due to Okta API error [{}]\n{}".format(
```

Comment:
login_error_code_parser(response["errorCode"], settings.okta_status_dict)
instead of 
error_string = "Exiting due to Okta API error [{}]\n{}".format(
            response['errorCode'], response['errorSummary'])
        if len(response['errorCauses']) > 0:
            error_string += "\n{}".format(json.dumps(response['errorCauses']))
        logging.error(error_string)